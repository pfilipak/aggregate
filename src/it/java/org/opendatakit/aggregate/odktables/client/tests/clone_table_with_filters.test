# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people
    name STRING false
    age INTEGER false
    weight DECIMAL false

insertSynchronizedRows user1 people 
    row1
        name Dylan
        age 22
        weight 175
    row2
        name John
        age 50
        weight 200
    row3
        name Jessica
        age 40
        weight 140

setTablePermissions user1 user2 people true false false

# Test basic filter
cloneSynchronizedTableWithFilters user2 people
    age GREATER_THAN 30
    
printTable user2 people
    
removeTableSynchronization user2 people

# Test removing synchronization removed filters
cloneSynchronizedTable user2 people

printTable user2 people

removeTableSynchronization user2 people

# Test somewhat complex filter
cloneSynchronizedTableWithFilters user2 people
    age GREATER_THAN_OR_EQUAL 40
    weight LESS_THAN_OR_EQUAL 140

printTable user2 people

# Test synchronize
insertSynchronizedRows user1 people
    row4
        name JessicasTwin
        age 40
        weight 140
    row5
        name AlmostJessicasTwin
        age 40
        weight 140.5
    row6
        name DefinitelyNotJessicasTwin
        age 40
        weight 170
        
synchronize user2 people

printTable user2 people

# Clean up
removeTableSynchronization user2 people
setTablePermissions user1 user2 people false false false
deleteSynchronizedTable user1 people
deleteUser user1
deleteUser user2