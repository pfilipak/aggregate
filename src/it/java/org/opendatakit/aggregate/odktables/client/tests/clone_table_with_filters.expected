# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people

insertSynchronizedRows user1 people 

setTablePermissions user1 user2 people true false false

# Test basic filter
cloneSynchronizedTableWithFilters user2 people
    
printTable user2 people
modification: 1
    row2
        age 50
        name John
        weight 200.0000000000
    row3
        age 40
        name Jessica
        weight 140.0000000000
    
removeTableSynchronization user2 people

# Test removing synchronization removed filters
cloneSynchronizedTable user2 people

printTable user2 people
modification: 1
    row1
        age 22
        name Dylan
        weight 175.0000000000
    row2
        age 50
        name John
        weight 200.0000000000
    row3
        age 40
        name Jessica
        weight 140.0000000000

removeTableSynchronization user2 people

# Test somewhat complex filter
cloneSynchronizedTableWithFilters user2 people

printTable user2 people
modification: 1
    row3
        age 40
        name Jessica
        weight 140.0000000000

# Test synchronize
insertSynchronizedRows user1 people
        
synchronize user2 people

printTable user2 people
modification: 2
    row3
        age 40
        name Jessica
        weight 140.0000000000
    row4
        age 40
        name JessicasTwin
        weight 140.0000000000

# Clean up
removeTableSynchronization user2 people
setTablePermissions user1 user2 people false false false
deleteSynchronizedTable user1 people
deleteUser user1
deleteUser user2
